app/services/get_stormglass_api_service.rb

```diff
@@ -10,18 +10,22 @@ def call
     lat = @spot.latitude;
     lng =  @spot.longitude;
     source = 'sg,meteo,noaa';
-    end_time = Time.now.utc.beginning_of_day + 7.days;
+    end_time = Time.now.utc.beginning_of_day + 1.days;
     params = 'waveHeight,swellDirection,swellHeight,swellPeriod,waveDirection,waveHeight,wavePeriod,windDirection,windSpeed,airTemperature,seaLevel,waterTemperature';
     response = HTTParty.get("https://api.stormglass.io/v2/weather/point?lat=#{lat}&lng=#{lng}&params=#{params}&source=#{source}&end=#{end_time}",
       headers: {
         'Authorization': "#{ENV['STORM_GLASS']}"
       }
     )
     rawresponse = JSON.parse(response.body)
-   #p rawresponse['meta']['requestCount'] # uncomment to display request count (max 50 /day)
     hours = rawresponse['hours']
+    # Calculate shakas more effectively
```

Comment:
i think i understand the shakas rating:
- the swell height rating is essentially the base rating as its the only positive one
- the base rating drops if the waves are too big
- the period has negative rating so reduces the base rating, but by less with increasing period
- the wind has negative rating so reduces the base rating, by more with increasing wind

This all makes sense to me. Looks good, there's no limit to how complex this rating system could get! I've added some suggestions below

Reply:
I think to be honest a weighting system that takes all into account is the most important for waves. personally i think the wind and the period are more important for deciing wae quality than the swell height. In which case make we could introduce a system where everthigng scores positively but there is a weighting for certain elements? Just an idea. this is far too complicated after half a bottle of red. It definitely works for now but am keen to discuss and develop this as its a key part of our app!